src/pgduckdb_node.cpp

```diff
@@ -92,25 +105,28 @@ ExecuteQuery(DuckdbScanState *state) {
 		ParamExternData tmp_workspace;
 
 		/* give hook a chance in case parameter is dynamic */
-		if (pg_params->paramFetch != NULL)
+		if (pg_params->paramFetch != NULL) {
 			pg_param = pg_params->paramFetch(pg_params, i + 1, false, &tmp_workspace);
-		else
+		} else {
 			pg_param = &pg_params->params[i];
+		}
 
 		if (pg_param->isnull) {
 			duckdb_params.push_back(duckdb::Value());
-		} else {
-			if (!OidIsValid(pg_param->ptype)) {
-				elog(ERROR, "parameter with invalid type during execution");
-			}
+		} else if (OidIsValid(pg_param->ptype)) {
 			duckdb_params.push_back(pgduckdb::ConvertPostgresParameterToDuckValue(pg_param->value, pg_param->ptype));
+		} else {
+			std::ostringstream oss;
+			oss << "parameter " << i << " has an invalid type (" << pg_param->ptype << ") during query execution";
+			throw duckdb::Exception(duckdb::ExceptionType::EXECUTOR, oss.str().c_str());
 		}
 	}
 
 	auto pending = prepared.PendingQuery(duckdb_params, true);
 	if (pending->HasError()) {
-		elog(ERROR, "DuckDB execute returned an error: %s", pending->GetError().c_str());
+		throw std::runtime_error(pending->GetError());
```

Comment:
Can use `pending->ThrowError()` instead

Reply:
That's what I did initially but sadly it throws an ugly JSON then... Maybe I missed something, I'll try again.

Reply:
I tried again and I got something like this:

```
ERROR:  (PGDuckDB/ExecuteQuery) {"exception_type":"Conversion","exception_message":"Could not convert string 'not an integer' to INT32\nLINE 1: SELECT (t)::integer AS t198 FROM public.foo\n                  ^","position":"10"}
```

Is there something specific to do so it doesn't output JSON?

Reply:
@Tishj I've pushed that [commit](https://github.com/duckdb/pg_duckdb/pull/251/commits/a91f2e636f08181febf0a621868d26f35fc5bd7d) to support the `pending->ThrowError()`. 
Maybe I missed something from DuckDB API, but I'm a bit surprised that:
* I have to specially handle `duckdb::Exception`
* `ThrowError` (in some cases at least) throws a `std::exception`, not a  `duckdb::Exception` one

Happy to improve if there's a better way?